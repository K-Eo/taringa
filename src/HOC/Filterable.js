import React, { Component } from 'react'
import { connect } from 'react-redux'

import { set } from '../filters/actions'

const withFilter = (filters = {}) => WrappedComponent => {
  class Filterable extends Component {
    componentDidMount() {
      this.props.dispatch(set(filters))
    }

    render() {
      return <WrappedComponent {...this.props} />
    }
  }

  return connect()(Filterable)
}

export default withFilter
