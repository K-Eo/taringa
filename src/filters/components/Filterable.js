import { Component } from 'react'
import { connect } from 'react-redux'

import { set } from '../actions'

export class Filterable extends Component {
  componentDidMount() {
    this.props.onDidMount()
  }

  render() {
    return null
  }
}

Filterable.defaultProps = {
  onDidMount: () => {},
}

const mapDispatchToProps = (dispatch, ownProps) => {
  const filters = ownProps.filters || {}
  return {
    onDidMount: () => dispatch(set(filters)),
  }
}

export default connect(null, mapDispatchToProps)(Filterable)
