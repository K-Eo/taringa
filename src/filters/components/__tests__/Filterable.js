import React from 'react'

import { Filterable } from '../Filterable'

describe('Filterable', () => {
  it('exists', () => {
    expect(Filterable).toBeDefined()
  })

  it('renders null', () => {
    const wrapper = shallow(<Filterable />)

    expect(wrapper).toMatchSnapshot()
  })

  it('calls onDidMount', () => {
    const mock = jest.fn()

    mount(<Filterable onDidMount={mock} />)

    expect(mock).toHaveBeenCalled()
  })
})
