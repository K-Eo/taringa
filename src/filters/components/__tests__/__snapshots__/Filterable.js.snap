// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filterable renders null 1`] = `""`;
